core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -150,7 +150,10 @@ private[spark] object JettyUtils extends Logging {
       private def doRequest(request: HttpServletRequest, response: HttpServletResponse): Unit = {
         beforeRedirect(request)
         // Make sure we don't end up with "//" in the middle
-        val newUrl = new URL(new URL(request.getRequestURL.toString), prefixedDestPath).toString
+        val requestURL = new URI(request.getRequestURL.toString).toURL
+        // scalastyle:off URLConstructor
+        val newUrl = new URL(requestURL, prefixedDestPath).toString
```

Comment:
This is a little tricky to change. So, this PR allows this as the only exception.

Reply:
```scala
val requestURI = new URI(request.getRequestURL.toString)
val newUri = requestURI.resolve(prefixedDestPath).toString
response.sendRedirect(newUri)
```
Will there be any problems with this change?

Reply:
Let me check. Thanks, @LuciferYang .

Reply:
To @LuciferYang , I checked multiple corner cases. And, it's not the same in some cases like the following. I'd prefer to keep the existing logic to avoid any risk here.

```
$ jshell
|  Welcome to JShell -- Version 21.0.4
|  For an introduction type: /help intro

jshell> new URL(new URL("https://a"), "?index.html")
$1 ==> https://a/?index.html

jshell> new URI("https://a").resolve("?index.html")
$2 ==> https://a?index.html
```

Reply:
To be clear, this PR aims to prevent any future addition of `new URL(` instead of removing everyone instances of `new URL(`.

Reply:
OK